### ISSUE TITLE
Make IP option dynamic

### ISSUE THREAD

**Issue by blacksponge** [zonemaster/zonemaster-gui#217]
I have installed zonemaster on a IPv4 only server so I disabled the IPv6 option in the profile but if I understand correctly the GUI default to true on both IPv4 and IPv6 without checking the value in profile. This cause the test to be executed for both each time, resulting in connectivity issue (https://zonemaster.ppsfleet.navy/result/82f4deea1693fc5b). I suggest to add a method in the backend so the client can fetch the needed settings from the profiles and fill the GUI's defaults according to those values.

**Comment by matsduf** [comment]
When the test is started the RPC call have three alternatives for `IPv4` and `IPv6`:

* true
* false
* (unset)

Both `true` and `false` will override what the profile says. If left unset then the setting in the profile will rule.

GUI will always set those to `true` or `false`, never unset. A reasonable change would be that GUI by default leaves those unset, and that the user can choose to set it to `true` or `false`.

```
IPv4: [x] default  [ ] yes  [ ] no  ||  IPv6: [x] default  [ ] yes  [ ] no
```

**Comment by blacksponge** [comment]
This is quick fix indeed, the drawback I see is that the user doesn't know what the default is and that we may end up in situation where both ipv4 and ipv6 are disabled. I still implemented it here #218 as this is still an improvement.

**Comment by matsduf** [comment]
I have given this issue a second thought. I do not think that the three-way selection on both IPv4 and IPv6 that I proposed above in https://github.com/zonemaster/zonemaster-gui/issues/217#issuecomment-858662304 is a good idea. It increases the complexity for the users to solve very rare use cases. With that I do not think that the present solution is good either.

For most users the GUI sits in front of a backend with both IPv4 and IPv6 connection and with both IPv4 and IPv6 being enabled in the profile. Turning on IPv4 or IPv6 has then no meaning since that is the default. Turning off IPv4 or IPv6 (but not both) could be meaningful for troubleshooting. What we need are buttons to be able to disable IPv4 or IPv6.

I rare situations we have a GUI in front of a backend with IPv4 only or IPv6 only. Then the other network can be disabled in the profile. In such a case buttons that can turn the networks off have no meaning, but they do not interfere with the profile setting, as the current solution.

The use cases where someone wants to set to `true` where it is `false` in the profile are very rare. They could be supported by running CLI on local machine instead.

My updated proposal is that we change the GUI so that it by default leaves both IPv4 and IPv6 setting to the RPC API unset, and set it to `false` if the equivalent button is pressed:
```
[ ] Disable IPv4  [ ] Disable IPv6
```
If possible, it should only be possible to select one. If the next is selected, then the first is unselected. If the RPCAPI receives `false` on both IPv4 and IPv6, then both are set to `true`.

@blacksponge, @pnax, @sandoche2k, @mattias-p and @vlevigneron, please read and comment!

**Comment by mattias-p** [comment]
I agree with the general complaint here, that users get unfairly punished for forgetting to disable IPv6 every time they start a test. But I don't think it's great to have an extra option in the UI that represents something but the exact meaning is opaque to the end-user.

I think it would make sense to have a configuration option in GUI that lets the administrator force disable the IPv6 button. And another option for IPv4 in case you're on an IPv6-only machine/network.

**Comment by matsduf** [comment]
@mattias-p, I am not sure what you suggest. What I suggest above could easily be implemented and would resolve the original issue and would also make it possible to disable IPv4 or IPv6 network.

There is no configuration concept in GUI. Everything is hard coded, or it could pick things up from Backend. That is done for the profiles. That requires more coding. We could start with the simple solution first.

**Comment by matsduf** [comment]
> This is quick fix indeed, the drawback I see is that the user doesn't know what the default is and that we may end up in situation where both ipv4 and ipv6 are disabled.

In such a case, which is probably unusual, there is a message, "Both IPv4 and IPv6 are disabled", from the Backend.

**Comment by mattias-p** [comment]
What I'm suggesting is that we implement a configuration feature in GUI that allows the administrator to adapt how the GUI works in certain ways. In this case the configuration would control what options are available to the end user regarding IPv4 and IPv6.

**Comment by blacksponge** [comment]
> `[ ] Disable IPv4  [ ] Disable IPv6`

This seems cleaner indeed. I've updated the PR accordingly.

I also agree on having a configuration feature, this would also help on other issues, as @mattias-p  already mentioned.

**Comment by blacksponge** [comment]
Resolved by #218
